library/src/main/java/bio/terra/landingzone/library/landingzones/definition/factories/CromwellStepsDefinitionProvider.java

```diff
@@ -61,82 +63,59 @@ public List<DefinitionVersion> availableVersions() {
   @Override
   public List<Pair<Step, RetryRule>> get(
       ArmManagers armManagers,
-      ParametersResolver parametersResolver,
+      ParametersResolverProvider parametersResolverProvider,
       ResourceNameProvider resourceNameProvider,
       LandingZoneProtectedDataConfiguration landingZoneProtectedDataConfiguration) {
     return List.of(
+        Pair.of(new GetManagedResourceGroupInfo(armManagers), RetryRules.cloud()),
+        Pair.of(new GetParametersResolver(parametersResolverProvider), RetryRules.shortDatabase()),
         Pair.of(
             new ValidateLandingZoneParametersStep(
                 InputParametersValidationFactory.buildValidators(
-                    StepsDefinitionFactoryType.CROMWELL_BASE_DEFINITION_STEPS_PROVIDER_TYPE),
-                parametersResolver),
+                    StepsDefinitionFactoryType.CROMWELL_BASE_DEFINITION_STEPS_PROVIDER_TYPE)),
             RetryRules.shortExponential()),
-        Pair.of(new GetManagedResourceGroupInfo(armManagers), RetryRules.cloud()),
         Pair.of(
-            new CreateNetworkSecurityGroupStep(
-                armManagers, parametersResolver, resourceNameProvider),
+            new CreateNetworkSecurityGroupStep(armManagers, resourceNameProvider),
             RetryRules.cloud()),
+        Pair.of(new CreateVnetStep(armManagers, resourceNameProvider), RetryRules.cloud()),
```

Comment:
Why did this used to be grouped with `CreateLogAnalyticsWorkspaceStep`? I don't understand how `Pair.of` was/is being used.

Reply:
They weren't grouped, they're just sequential. The `Pair.of` is used to bundle the step with its retry rule which `addCreateSteps` in `CreateLandingZoneResourcesFlight` ([link](https://github.com/DataBiosphere/terra-landing-zone-service/blob/1f30f3886db5c74b73146a64dc7fa7abbdaa195b/library/src/main/java/bio/terra/landingzone/stairway/flight/create/CreateLandingZoneResourcesFlight.java#L76)) uses to construct the flight

Reply:
Ah, I see… I was just misreading the diff. Thanks!